lib/bango/views/billing.py

```diff
@@ -0,0 +1,144 @@
+from django.conf import settings
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from lib.bango.client import BangoError, get_client
+from lib.bango.constants import MICRO_PAYMENT_TYPES, PAYMENT_TYPES
+from lib.bango.errors import ProcessError
+from lib.bango.forms import CreateBillingConfigurationForm
+from lib.bango.serializers import SellerProductBangoOnly
+from lib.bango.utils import sign
+from lib.bango.views.base import BangoResource
+from solitude.constants import PAYMENT_METHOD_OPERATOR
+from solitude.logger import getLogger
+
+log = getLogger('s.bango')
+
+
+def prepare(form):
+    data = form.bango_data
+
+    # Used to create the approprate data structure.
+    client = get_client()
+    billing = client.client('billing')
+    price_list = billing.factory.create('ArrayOfPrice')
+    price_types = set()
+
+    for item in form.cleaned_data['prices']:
+        price = billing.factory.create('Price')
+        price.amount = item.cleaned_data['price']
+        price.currency = item.cleaned_data['currency']
+        price_types.add(item.cleaned_data['method'])
+
+        # TODO: remove this.
+        # Very temporary and very fragile hack to fix bug 882183.
+        # Bango cannot accept regions with price info so if there
+        # are two USD values for different regions it triggers a 500 error.
+        append = True
+        for existing in price_list.Price:
+            if existing.currency == price.currency:
+                log.info('Skipping %s:%s because we already have %s:%s'
+                         % (price.currency, price.amount,
+                            existing.currency, existing.amount))
+            append = False
+            break
+
+    if append:
```

Comment:
the old code was appending within the for loop. I think you still want to do that, no?